cli/src/lib.rs

```diff
@@ -583,6 +588,34 @@ fn new_program(name: &str) -> Result<()> {
     Ok(())
 }
 
+fn rename_program(from: &str, to: &str) -> Result<()> {
+    rename_in_file(&Path::new(&format!("programs/{}/Cargo.toml", from)), &from, &to)?;
+    rename_in_file(&Path::new(&format!("programs/{}/src/lib.rs", from)), &from, &to)?;
+    fs::rename(&format!("programs/{}", from), &format!("programs/{}", to))?;
+    
+    // Users probably have either .js or .ts tests, not both. Don't panic.
+    rename_in_file(&Path::new(&format!("tests/{}.js", from)), &from, &to).ok();
+    rename_in_file(&Path::new(&format!("tests/{}.ts", from)), &from, &to).ok();
+    fs::rename(&format!("tests/{}.js", from), &format!("tests/{}.js", to)).ok();
+    fs::rename(&format!("tests/{}.ts", from), &format!("tests/{}.ts", to)).ok();
+
+    Ok(())
+}
+
+fn rename_in_file(path: &Path, from: &str, to: &str) -> Result<()> {
```

Comment:
should we just use all available cases from `heck` here?

Reply:
yeah i'll add them. don't see any drawbacks to it besides more processing (should be negligible though)

Reply:
should we bother upgrading to `heck 0.4.0`? imo, i'm a bit hesitant as there are some breaking changes to `cli/src/config.rs` and `cli/src/template.rs`.

Reply:
are there any user facing breaking changes?

Reply:
if not, feel free to upgrade. I like getting rid of `mixed_case`

Reply:
sg! agree, nothing user facing

Reply:
tested this a bit more and there's a gotcha im seeing:

1 word to 2+ word conversion. undefined on which output it should result in. for example:
running `anchor rename apple applePie`
```
upper camel: Apple to ApplePie
lower camel: apple to applePie
snake: apple to apple_pie
kebab: apple to apple-pie
shouty snake: APPLE to APPLE_PIE
shouty kebab: APPLE to APPLE-PIE
```
there are 3 possible outputs for `apple` here. imo users will most likely stick to what's given in `template.rs` so maybe we account for those conversions only? wdyt?

Reply:
wdyt about rejecting user inputs of 1 word-length names to 2 word-length names? 
for example:
`anchor rename apple apple-pie`
gets a bit ugly..
```
upper camel: Apple to ApplePie
lower camel: apple to applePie
snake: apple to apple_pie
kebab: apple to apple-pie
shouty snake: APPLE to APPLE_PIE
shouty kebab: APPLE to APPLE-PIE
```
kebab, snake, lower camel are all possibilities here

Reply:
for the lolz, check out the effects of rename chaining 😆 
```
import * as anchor from "@project-serum/anchor";
import { Program } from "@project-serum/anchor";
import { Pasta } from "../target/types/pasta";

describe("pasta", () => {
  // Configure the client to use the local cluster.
  anchor.setProvider(anchor.Provider.env());

  const program = anchor.workspace.Pasta as Program<Pasta>;

  it("Is initialized!", async () => {
    // Add your test here.
    const tx = await program.rpc.initialize({});
    console.log("Your transaction signature", tx);
  });
});
```
`anchor rename pasta pasta-alfredo`
```
import * as anchor from "@project-serum/anchor";
import { Program } from "@project-serum/anchor";
import { PastaAlfredo } from "../target/types/pasta_alfredo-alfredoAlfredo";

describe("pasta_alfredo-alfredoAlfredo", () => {
  // Configure the client to use the local cluster.
  anchor.setProvider(anchor.Provider.env());

  const program = anchor.workspace.PastaAlfredo as Program<PastaAlfredo>;

  it("Is initialized!", async () => {
    // Add your test here.
    const tx = await program.rpc.initialize({});
    console.log("Your transaction signature", tx);
  });
});

```

Reply:
ahhh im braindead. lower camel is crucial for the IDL.

Reply:
> wdyt about sticking to upper camel, snake, and shouty snake? 
ahhh im braindead. lower camel is crucial for the IDL.

anyways.. i will try [aho-corasick](https://docs.rs/aho-corasick/0.7.18/aho_corasick/#example-replacing-matches-in-a-stream) out sometime to address the chaining issue above.